src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImportPowerShellDataFile.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// This class implements Import-PowerShellDataFile command.
+    /// </summary>
+    [Cmdlet(VerbsData.Import, "PowerShellDataFile", DefaultParameterSetName = "ByPath",
+        HelpUri = "https://go.microsoft.com/fwlink/?LinkID=623621", RemotingCapability = RemotingCapability.None)]
+    public class ImportPowerShellDataFileCommand : PSCmdlet
+    {
+        bool _isLiteralPath;
+
+        /// <summary>
+        /// Path specified, using globbing to resolve
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = "ByPath",
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true)]
+        [Alias("PSPath")]
+        [ValidateNotNullOrEmpty()]
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
+        public string[] Path { get; set; }
+
+        /// <summary>
+        /// Specifies a path to one or more locations, without globbing
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = "ByLiteralPath")]
+        [ValidateNotNullOrEmpty()]
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
+        public string[] LiteralPath
+        {
+            get { return Path; }
+            set { _isLiteralPath = true; Path = value; }
+        }
+
+        /// <summary>
+        /// For each path, resolve it, parse it and write all hashtables
+        /// to the output stream
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            foreach (var path in Path)
+            {
+                var resolved = PathUtils.ResolveFilePath(path, this, _isLiteralPath);
+                if(!string.IsNullOrEmpty(resolved) && System.IO.File.Exists(resolved))
+                {
+                    Token[] tokens;
+                    ParseError[] errors;
+                    var ast = Parser.ParseFile(resolved, out tokens, out errors);
+                    if (errors.Length > 0)
+                    {
+                        WriteInvalidDataFileError(resolved, String.Empty);
+                    }
+                    else
+                    {
+                        var data = ast.Find(a => a is HashtableAst, false);
```

Comment:
I realize this is just a translation of the PowerShell implementation, but a psd1 can contain much more than a simple hashtable.  For example, the psd1 file contain 2 hashtables as an array, and this implementation would ignore the second one.

As implemented, the cmdlet is mostly useful for typical module manifests, but psd1 files are used for other things, e.g. localization, and sometimes use commands like `ConvertFrom-StringData`.

We should, as a minimum, not ignore any part of the ast.
We should also consider supporting more than a simple hashtable. `Ast.SafeGetValue` can certainly handle more (e.g. an array of hashtables).
We might also consider doing something more like what `Import-LocalizedData` does, though we'd need feedback from @LeeHolmes to understand the important differences and why this cmdlet was introduced.